### ISSUE TITLE
TypeError: Not an Operation or Iterable: <class 'cirq.ops.global_phase_op.GlobalPhaseGate'> (-0.2272020946930871+0.9738476308781951j)

### ISSUE THREAD

**Issue by ACE07-Sev** [quantumlib/Cirq#6558]
Greetings,

Hope all are well. I am trying to add a Global Phase Gate to the circuit definition, however I am receiving the error below.
```
TypeError: Not an Operation or Iterable: <class 'cirq.ops.global_phase_op.GlobalPhaseGate'> (-0.2272020946930871+0.9738476308781951j)
```
Here is my code :
```
# Create a Global Phase gate
phase = np.exp(1j*angle)
global_phase = cirq.GlobalPhaseGate(phase)

# Apply the Global Phase gate to the circuit
circuit.append(global_phase)
```
Thanks in advance!

**Comment by ACE07-Sev** [comment]
Shouldn't global phase not depend on a specific qubit? It's an attribute of the circuit, not specific qubits no?

**Comment by pavoljuhas** [comment]
> Shouldn't global phase not depend on a specific qubit? It's an attribute of the circuit, not specific qubits no?

That is correct, sorry about misspeaking.  The circuit needs Operation-type objects and does not accept Gate instances.
For GlobalPhaseGate there are indeed no specific qubits as you correctly pointed out.
The gate thus needs to be called without arguments so it is expressed as an Operation.

Either of `cirq.GlobalPhaseGate(phase)()` or `cirq.GlobalPhaseGate(phase).on()` will do in your example.

**Comment by ACE07-Sev** [comment]
No worries at all. Thank you so much for the help. Quick question, I am trying to write a tester for this, and was wondering how I can check the global phase of a circuit.

**Comment by ACE07-Sev** [comment]
@pavoljuhas ?

**Comment by pavoljuhas** [comment]
> ... Quick question, I am trying to write a tester for this, and was wondering how I can check the global phase of a circuit.

Apologies about a delay in response.  cirq does not have tracking of global phase for circuits because the global phase can also change with gate operations - for example a gate sequence XZXZ is equivalent to a global phase shift by PI (multiplication by -1).  One option to check the effects of global phase is to evaluate the unitary matrix `cirq.unitary(circuit)` and verify the complex phase of its elements changes by equal amount as the global phase coefficient.  Hope this works.

BTW, w/r to the initial question, cirq has also a convenience function `cirq.global_phase_operation(phase)` to construct the phase operation without explicitly using the `GlobalPhaseGate`.

**Comment by ACE07-Sev** [comment]
I see. So, how would I let's say check if the global phase gate of e^i1.8 is correctly applied to the circuit in a tester function?

**Comment by NoureldinYosri** [comment]
@ACE07-Sev 

```py3
np.testing.assert_allclose(cirq.unitary(circuit_with_global_phase), cirq.unitary(circuit_without_global_phase) * global_phase)
```

although I don't see why you need to test that. testing that global phase is applied is equivalent to testing cirq itself, I don't think your goal is that -unless you are contribution to cirq- instead you should be testing the logic of your method/class in which case that would be checking that your method constructs a circuit with a global phase object that has the correct phase e.g.
```py3
global_phase = cirq.global_phase_operation(desired_global_phase)
circuit = my_method()
assert any(op == global_phase for op in circuit.all_operations())
```

but if you expect that their might be multiple global phase operations in your circuit, then you can do

```py3
total_phase = 1
for op in circuit.all_operations():
  if isinstance(op.gate, cirq.GlobalPhaseGate):
     total_phase *= op.gate.coefficient
assert total_phase == pytest.approx(desired_global_phase)
```

**Comment by ACE07-Sev** [comment]
Thank you so much!
